app/frontend/src/components/Alert.tsx

```diff
@@ -16,9 +18,28 @@ interface AlertProps extends MuiAlertProps {
   severity: MuiAlertProps["severity"];
 }
 
-export default function Alert({ className, ...otherProps }: AlertProps) {
+export default function Alert({
+  className,
+  children,
+  ...otherProps
+}: AlertProps) {
   const classes = useStyles();
+
   return (
-    <MuiAlert {...otherProps} className={classNames(classes.root, className)} />
+    <MuiAlert {...otherProps} className={classNames(classes.root, className)}>
+      {React.Children.map(children, (child) => {
+        if (typeof child !== "string")
+          throw new Error("Alert should only have string children");
+
+        // Search for the error in the ugly grpc error object keys
+        // Replace it with the nice error if found
+        const oldErrorKey = Object.keys(grpcErrorStrings).find((oldError) =>
+          child.includes(oldError)
+        );
+        if (oldErrorKey)
+          return grpcErrorStrings[oldErrorKey as keyof typeof grpcErrorStrings];
+        else return child;
+      })}
```

Comment:
This is gross but don't think there's a better way.

Reply:
Do we ever use `Alert` with more than one children? If not, and it's a bit crap for a generic component, but could type the `children` as string so we can operate on `children` directly to do this. Will simplify the code a fair bit (no need for `React.Children` and the error throwing bit etc)

Reply:
Yes quite a few places it has `<Alert>Some text and a {var}</Alert>` which I wanted to support, but now I think I will do as you say since it will probably be constants/I18n keys later anyway.